examples/lines_bars_and_markers/filled_step.py

```diff
@@ -15,6 +15,28 @@
 import matplotlib.ticker as mticker
 from cycler import cycler
 
+###############################################################################
+# Plain filled steps
+
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(9, 4.5), tight_layout=True)
+ax1.fill_between([0, 1, 2, 3], [1, 2, 3], step='between')
+ax2.fill_betweenx([0, 1, 2, 3], [0, 1, 2], step='between')
+ax1.set_ylabel('counts')
+ax1.set_xlabel('edges')
+ax2.set_xlabel('counts')
+ax2.set_ylabel('edges')
+
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(9, 4.5), tight_layout=True)
+ax1.fill_between([0, 1, 2, 3], [1, 2, 3], [0, 1, 0], step='between')
+ax2.fill_betweenx([0, 1, 2, 3], [1, 2, 3], [0, 1, 0], step='between')
+ax1.set_ylabel('counts')
+ax1.set_xlabel('edges')
+ax2.set_xlabel('counts')
+ax2.set_ylabel('edges')
+
+
+###############################################################################
+# Hatches
```

Comment:
Since the hatches histogram are much more eye-catchy we should probably still show them in the gallery, which would mean to add

     # sphinx_gallery_thumbnail_number = 2

somewhere in the code